### PR TITLE
[SPARK-44208][CORE][SQL] Assign clear error class names for some logic that directly uses exceptions

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to assign clear error class names for some logic that directly uses exceptions, include:
- ALL_PARTITION_COLUMNS_NOT_ALLOWED
- INVALID_HIVE_COLUMN_NAME
- SPECIFY_BUCKETING_IS_NOT_ALLOWED
- SPECIFY_PARTITION_IS_NOT_ALLOWED
- UNSUPPORTED_ADD_FILE.DIRECTORY
- UNSUPPORTED_ADD_FILE.LOCAL_DIRECTORY

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Update UT.
- Pass GA.

### CODE REVIEW

**Comment by panbingkun** [comment]
The pr is to fix the issue of `errorClass=null` mentioned by https://github.com/apache/spark/pull/41699

**Comment by panbingkun** [comment]
cc @MaxGekk

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1035,6 +1040,11 @@
       "The expected format is ByteString, but was <unsupported> (<class>)."
     ]
   },
+  "INVALID_COLUMN_NAME" : {
+    "message" : [
+      "Cannot create a table<tableName> having a nested column<columnName> whose name contains invalid characters <invalidChars> in Hive metastore."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1045]:
```suggestion
      "Cannot create the table <tableName> having the nested column <columnName> whose name contains invalid characters <invalidChars> in Hive metastore."
```

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -2264,6 +2284,23 @@
...
     ]
   },
+  "UNSUPPORTED_ADD_FILE" : {
+    "message" : [
+      "Don't support add file."
+    ],
+    "subClass" : {
+      "DIRECTORY" : {
+        "message" : [
+          "The file<path> is a directory, you must set \"spark.sql.legacy.addSingleFileInAddFile\" to \"false\"."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:2294]:
```suggestion
          "The file <path> is a directory, you must set \"spark.sql.legacy.addSingleFileInAddFile\" to \"false\"."
```

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -2264,6 +2284,23 @@
...
+        ]
+      },
+      "LOCAL_DIRECTORY" : {
+        "message" : [
+          "The file<path> is a local directory, but not running \"local mode\"."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:2299]:
```suggestion
          "The file <path> is a local directory, but not running \"local mode\"."
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json [hunk 4]
```diff
@@ -1035,6 +1040,11 @@
       "The expected format is ByteString, but was <unsupported> (<class>)."
     ]
   },
+  "INVALID_COLUMN_NAME" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1043]: This seems a Hive specific error. Could you add the word `HIVE` to the error class name like `INVALID_HIVE_COLUMN_NAME`

File: core/src/main/resources/error/error-classes.json [hunk 5]
```diff
@@ -4,6 +4,11 @@
       "Non-deterministic expression <sqlExpr> should not appear in the arguments of an aggregate function."
     ]
   },
+  "ALL_FOR_PARTITION_COLUMNS_IS_NOT_ALLOWED" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:7]: nit:
```suggestion
  "ALL_PARTITION_COLUMNS_NOT_ALLOWED" : {
```

File: core/src/main/resources/error/error-classes.json [hunk 6]
```diff
@@ -2264,6 +2284,23 @@
...
     ]
   },
+  "UNSUPPORTED_ADD_FILE" : {
+    "message" : [
+      "Don't support add file."
+    ],
+    "subClass" : {
+      "DIRECTORY" : {
+        "message" : [
+          "The file <path> is a directory, you must set \"spark.sql.legacy.addSingleFileInAddFile\" to \"false\"."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:2294]:
```suggestion
          "The file <path> is a directory, consider to set \"spark.sql.legacy.addSingleFileInAddFile\" to \"false\"."
```

File: core/src/main/resources/error/error-classes.json [hunk 7]
```diff
@@ -2264,6 +2284,23 @@
...
+        ]
+      },
+      "LOCAL_DIRECTORY" : {
+        "message" : [
+          "The file <path> is a local directory, but not running \"local mode\"."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:2299]:
```suggestion
          "The local directory <path> is not supported in a non-local master mode."
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by MaxGekk** [comment]
@panbingkun Could you resolve conflicts, please.

File: core/src/main/resources/error/error-classes.json (hunk 7 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json:2299]: I really like this change

**Comment by panbingkun** [comment]
> @panbingkun Could you resolve conflicts, please.

Let me update it.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
@panbingkun Could you update PR description (error class names) according to your recent changes.

**Comment by panbingkun** [comment]
> @panbingkun Could you update PR description (error class names) according to your recent changes.

Okay

**Pull request closed without merging**